pkg/webhook/pod_webhook.go

```diff
@@ -17,62 +17,219 @@ limitations under the License.
+	"fmt"
+	"strings"
 
 	kueue "github.com/openshift/kueue-operator/pkg/apis/kueueoperator/v1alpha1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// arrays for the list of webhooks we remove if pod integration is not needed
-// Kueue builds statefulset and deployment off on pod integration
+// kueueManagedLabel is the label used to opt-in resources for kueue.
+const kueueManagedLabel = "kueue.openshift.io/managed"
+
 var (
-	podBasedValidatingWebhooks = []string{"vdeployment.kb.io", "vpod.kb.io", "vstatefulset.kb.io"}
-	podBasedMutatingWebhooks   = []string{"mdeployment.kb.io", "mpod.kb.io", "mstatefulset.kb.io"}
+	// webhooksList maps the webhook name to the framework/resource name.
+	// for e.g. "BatchJob" -> (vjob.kb.io -> job)
+	webhooksList = map[string]string{
+		string(kueue.KueueIntegrationBatchJob):    "job",
+		string(kueue.KueueIntegrationMPIJob):      "mpijob",
+		string(kueue.KueueIntegrationRayJob):      "rayjob",
+		string(kueue.KueueIntegrationRayCluster):  "raycluster",
+		string(kueue.KueueIntegrationJobSet):      "jobset",
+		string(kueue.KueueIntegrationPaddleJob):   "paddlejob",
+		string(kueue.KueueIntegrationPyTorchJob):  "pytorchjob",
+		string(kueue.KueueIntegrationTFJob):       "tfjob",
+		string(kueue.KueueIntegrationXGBoostJob):  "xgboostjob",
+		string(kueue.KueueIntegrationPod):         "pod",
+		string(kueue.KueueIntegrationDeployment):  "deployment",
+		string(kueue.KueueIntegrationStatefulSet): "statefulset",
+		"kueue.x-k8s.io/clusterqueue":             "clusterqueue",
+		"kueue.x-k8s.io/workload":                 "workload",
+		"kueue.x-k8s.io/resourceflavor":           "resourceflavor",
+		"kueue.x-k8s.io/cohort":                   "cohort",
+	}
 )
 
 func ModifyPodBasedValidatingWebhook(kueueCfg kueue.KueueConfiguration, currentWebhook *admissionregistrationv1.ValidatingWebhookConfiguration) *admissionregistrationv1.ValidatingWebhookConfiguration {
-	// if there is a pod based webhook, we need to safely enabled it
-	// For now we will not modify this
-	for _, val := range kueueCfg.Integrations.Frameworks {
-		if val == kueue.KueueIntegrationPod || val == kueue.KueueIntegrationDeployment || val == kueue.KueueIntegrationStatefulSet {
-			return currentWebhook
-		}
+	newWebhook := currentWebhook.DeepCopy()
+	newWebhook.Webhooks = []admissionregistrationv1.ValidatingWebhook{}
+
+	enabledFrameworks := make(map[string]bool)
+	podSelector := defaultLabelSelector()
+
+	for _, fw := range kueueCfg.Integrations.Frameworks {
+		enabledFrameworks[string(fw)] = true
 	}
-	newWebHook := currentWebhook.DeepCopy()
-	newWebHook.Webhooks = []admissionregistrationv1.ValidatingWebhook{}
 
-	for _, val := range currentWebhook.Webhooks {
-		if !findWebhook(val.Name, podBasedValidatingWebhooks) {
-			newWebHook.Webhooks = append(newWebHook.Webhooks, val)
+	for _, wh := range currentWebhook.Webhooks {
+		framework := getFrameworkForWebhook(wh.Name, "v")
```

Comment:
"v" and "w" could be more explicit... getFrameworkForValidatingWebhook and getFrameworkForMutatingWebhook. Thoughts?

Reply:
The primary reason for keeping it this way is to avoid a few extra lines, but I'm not opposed to this idea